### ISSUE TITLE
Error 422 on late bytes upload

### ISSUE THREAD

**Issue by megascatterbomb** [MegaAntiCheat/client-backend#175]
Several users have reported intermittent 422 errors when uploading the late bytes.

![image](https://github.com/user-attachments/assets/5c3cf4fe-c968-437f-9a67-200b9bf21d37)

Not critical as it doesn't affect our ability to parse the demo (it can even be played back in TF2 still), but still worth investigating.

**Comment by jayceslesar** [comment]
I am seeing this a bunch -- seems wack 
```
"POST /late_bytes?api_key=some_api_key HTTP/1.1" 422 Unprocessable Entity
```

**Comment by jayceslesar** [comment]
I think if we make this logic better in terms of verbosity when it matters we will get some more info https://github.com/MegaAntiCheat/masterbase/blob/main/masterbase/app.py#L383

**Comment by jayceslesar** [comment]
there seems to be no rhyme or reason with patterns for a given api key -- some will succeed and others will fail

**Comment by JoshuaVandaele** [comment]
Could this be related to the "Auto-delete recordings" option?
![image](https://github.com/user-attachments/assets/ed2150f0-3dde-4aee-95d2-4838cc909405)

**Comment by megascatterbomb** [comment]
> Could this be related to the "Auto-delete recordings" option? ![image](https://private-user-images.githubusercontent.com/24195176/354415423-ed2150f0-3dde-4aee-95d2-4838cc909405.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1NTYyODAsIm5iZiI6MTcyMjU1NTk4MCwicGF0aCI6Ii8yNDE5NTE3Ni8zNTQ0MTU0MjMtZWQyMTUwZjAtM2RkZS00YWVlLTk1ZDItNDgzOGNjOTA5NDA1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAxVDIzNDYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5MTI2NTBmNGQ0N2E3YmNkYjIzYjhmYzA5ZDdmN2Q3OWIyOTA2ZGU1OWNkNDQ3YmNjNmY3NzhiOWIxMzQwYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xONh0kdSX46Z210je4qgMMjT-30JnhrFYRPMg4BZfvs)

I've encountered this error a few times myself and I don't have that setting checked.

**Comment by jayceslesar** [comment]
I am struggling to reproduce locally

**Comment by jayceslesar** [comment]
The only way it would fail validation is if the body isnt a string

**Comment by jayceslesar** [comment]
reference https://github.com/MegaAntiCheat/client-backend/blob/main/src/masterbase.rs#L232

**Comment by jayceslesar** [comment]
This smells like a race condition on the DB/filesystem side as I cannot reproduce with what I believe to be valid data

**Comment by jayceslesar** [comment]
note that we return a 422 if we get an error when attempting to close the late bytes out ... https://github.com/MegaAntiCheat/masterbase/blob/main/masterbase/app.py#L128

likely we are hitting https://github.com/MegaAntiCheat/masterbase/blob/main/masterbase/lib.py#L559 but I just added logs to show more detail

**Comment by jayceslesar** [comment]
yes is indeed a race condition on the backend. Closing this issue and will eventually reproduce and fix

**Comment by Familex** [comment]
got these errors too:

```
2024-08-13T05:26:32.011520Z  INFO client_backend::demo: Opened new demo session with Masterbase: DemoSession(229295141473228488338479370295583856718)
2024-08-13T06:04:35.083956Z ERROR client_backend::demo: Failed to upload late bytes to masterbase: Server returned 422
2024-08-13T06:04:37.021542Z  INFO client_backend::masterbase: Closed session 229295141473228488338479370295583856718.
2024-08-13T06:05:08.936735Z  INFO client_backend::demo: Opened new demo session with Masterbase: DemoSession(312197181810514410830725529466809385660)
2024-08-13T06:31:51.085087Z ERROR client_backend::demo: Failed to upload late bytes to masterbase: Server returned 422
2024-08-13T06:31:52.835987Z  INFO client_backend::masterbase: Closed session 312197181810514410830725529466809385660.
```
